src/components/WelcomeScreen.js

```diff
@@ -0,0 +1,35 @@
+import React from 'react';
+
+class WelcomeScreen extends React.Component {
+    constructor(props) {
+      super(props);
+      this.state = {value: ''};
+      this.handleSubmit = this.handleSubmit.bind(this);
+    }
+
+    handleSubmit (e) {
+      e.preventDefault();
+      console.log(this.input.value);
+      this.props.setUsername(this.input.value);
```

Comment:
ah, ok I am being thick, this value should actually be in the props for your component, ie the value of this input should live inside redux state.. Its a massive pain / problem with react / redux so we can leave it for now.